### DISCUSSION TITLE
How to For loop with useContractRead

### DISCUSSION THREAD

**Discussion by kriptoe** [wevm/wagmi#2063] [General]
I want to do a for loop to call contractRead to get all the NFTs owned by a user for a single contract.
I know how to do a single call to useContractRead but don't know how to loop through it several times with a different ownerID each time

const { data: nftTokens, error: getNFTError } = useContractRead({
    ...contractConfig,
    functionName: "tokenOfOwnerByIndex",
    args: ["0xa435530d50d7D17Fd9fc6E1c897Dbf7C08E12d35",ownerID], //hardcoded address can create a state variable
  });

I tried this but it just gets the same NFT ID every time it loops
    for (let i=0; i < 5; i++) {
 try{
      alert(i + " " + nftTokens.toString())
      setOwnerID(i)
     let addr = "https://ipfs.io/ipfs/QmSg4CMhmWdQ17i7pNbd8ENhW3B4Vb1kvMK3pgj7tryaNv/" + nftTokens.toString() + ".jpg";
     collectibleUpdate.push({ id: nftTokens.toString(), image: addr, owner: address});
      }

**Comment by murderteeth** [comment]
Have you checked out infinite reads? https://wagmi.sh/react/hooks/useContractInfiniteReads
